undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentProcessor.java

```diff
@@ -404,15 +416,50 @@ public Duration getDefaultSessionTimeout() {
                     }
                 };
                 CapabilityServiceConfigurator factoryConfigurator = provider.getSessionManagerFactoryServiceConfigurator(factoryServiceName, configuration);
-                CapabilityServiceConfigurator codecConfigurator = provider.getSessionIdentifierCodecServiceConfigurator(codecServiceName, configuration);
-
                 sessionManagerFactory = builder.requires(factoryConfigurator.getServiceName());
-                sessionIdentifierCodec = builder.requires(codecConfigurator.getServiceName());
-
                 factoryConfigurator.configure(capabilitySupport).build(serviceTarget).install();
-                codecConfigurator.configure(capabilitySupport).build(serviceTarget).install();
+
+                // Skip route locator service installation in non-distributable case
+                if (provider.getRouteLocatorServiceConfigurator(configuration) != null) {
+                    provider.getRouteLocatorServiceConfigurator(configuration).configure(capabilitySupport).build(serviceTarget).install();
```

Comment:
Why create this CapaabilityServiceConfigurator twice?

Reply:
Fixed.